Superfluous ignore comments warnings catch 22

**Describe the bug**

For certain protocol properties, if I use `// periphery:ignore` on them, `periphery scan` outputs a "Superfluous ignore comment for property" warning, but if I remove the ignore comments, it outputs an "Unused property" warning.

**Reproduction**

See the difference for the warnings for lines 2 & 3 in the output of `periphery scan --relative-results` (you can remove or add the ignore comments to see them switch back & forth):

```console
$ periphery scan --relative-results
* Building...
* Indexing...
* Analyzing...

Sources/mas/main.swift:2:6: warning: Unused property 'a'
Sources/mas/main.swift:3:6: warning: Superfluous ignore comment for property 'b' (declaration is referenced and should not be ignored)
Sources/mas/main.swift:7:6: warning: Unused property 'ai'
Sources/mas/main.swift:12:6: warning: Unused property 'a'
Sources/mas/main.swift:22:6: warning: Unused property 'ai'
```

For the example `main.swift`:

```swift
protocol AB {
	var a: String { get }
	var b: String { get } // periphery:ignore
}

protocol ABI: AB {
	var ai: Int { get }
	var bi: Int { get }
}

extension ABI {
	var a: String {
		.init(ai)
	}

	var b: String {
		.init(bi)
	}
}

struct ABIS: ABI {
	let ai = 1
	let bi = 0
}

print(ABIS())
```

Example `Package.swift`:

```swift
// swift-tools-version:6.2

private import PackageDescription

_ = Package(
	name: "mas",
	platforms: [.macOS(.v13)],
	products: [.executable(name: "mas", targets: ["mas"])],
	targets: [.executableTarget(name: "mas")],
	swiftLanguageModes: [.v6],
)
```

**Environment**

```console
$ periphery version
3.5.1
$ swift -version
swift-driver version: 1.127.14.1 Apple Swift version 6.2.3 (swiftlang-6.2.3.3.21 clang-1700.6.3.2)
Target: arm64-apple-macosx26.0
$ xcodebuild -version
Xcode 26.2
Build version 17C52
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@
 ##### Bug Fixes
 
 - Fix some causes of potentially non-deterministic results.
+- Fix incorrect superfluous ignore comment results for protocol members.
 
 ## 3.5.1 (2026-02-15)
 

--- a/Sources/PeripheryKit/ScanResultBuilder.swift
+++ b/Sources/PeripheryKit/ScanResultBuilder.swift
@@ -90,12 +90,17 @@ public enum ScanResultBuilder {
     /// This indicates that the declaration would have been marked as used even without the ignore command.
     private static func hasReferencesFromNonIgnoredCode(_ decl: Declaration, graph: SourceGraph) -> Bool {
         let references = graph.references(to: decl)
+        let isProtocolMember = decl.parent?.kind == .protocol
 
         for ref in references {
             guard ref.kind != .retained, let parent = ref.parent else { continue }
 
+            // Protocol members naturally accumulate related references from conformances and
+            // default implementations even when genuinely unused. Only normal references indicate
+            // actual usage.
+            if isProtocolMember, ref.kind != .normal { continue }
+
             if graph.commandIgnoredDeclarations[parent] == nil {
-                // Check that the parent is actually used (not itself unused)
                 if graph.usedDeclarations.contains(parent) {
                     return true
                 }

--- a/Tests/Fixtures/Sources/RetentionFixtures/testSuperfluousIgnoreCommandOnProtocolMember.swift
+++ b/Tests/Fixtures/Sources/RetentionFixtures/testSuperfluousIgnoreCommandOnProtocolMember.swift
@@ -1,0 +1,37 @@
+import Foundation
+
+// MARK: - Protocol member with ignore that is NOT superfluous
+// The member only has related references from conformances/default implementations,
+// not normal references indicating actual usage.
+
+public protocol CorrectlyIgnoredProtocol {
+    var ignoredProperty: String { get } // periphery:ignore
+}
+
+public extension CorrectlyIgnoredProtocol {
+    var ignoredProperty: String { "" }
+}
+
+public class CorrectlyIgnoredProtocolConformingClass {
+    public let ignoredProperty = 0
+}
+
+// MARK: - Protocol member with ignore that IS superfluous
+// The member has a normal reference (it's actually called), so the ignore is superfluous.
+
+public protocol SuperfluouslyIgnoredProtocol {
+    var superfluousProperty: String { get } // periphery:ignore
+}
+
+extension SuperfluouslyIgnoredProtocol {
+    var superfluousProperty: String { "" }
+}
+
+public class SuperfluouslyIgnoredProtocolConformingClass: SuperfluouslyIgnoredProtocol {
+    public var superfluousProperty: String { "" }
+}
+
+public func useSuperfluousProtocolMember() {
+    let instance: SuperfluouslyIgnoredProtocol = SuperfluouslyIgnoredProtocolConformingClass()
+    _ = instance.superfluousProperty
+}

--- a/Tests/PeripheryTests/RetentionTest.swift
+++ b/Tests/PeripheryTests/RetentionTest.swift
@@ -1236,6 +1236,22 @@ final class RetentionTest: FixtureSourceGraphTestCase {
         }
     }
 
+    func testSuperfluousIgnoreCommandOnProtocolMember() {
+        analyze(retainPublic: true) {
+            // Protocol member with ignore that only has related references (from conformances
+            // and default implementations) - the ignore is NOT superfluous.
+            assertReferenced(.protocol("CorrectlyIgnoredProtocol")) {
+                self.assertNotSuperfluousIgnoreCommand(.varInstance("ignoredProperty"))
+            }
+
+            // Protocol member with ignore that has normal references (actually used) -
+            // the ignore IS superfluous.
+            assertReferenced(.protocol("SuperfluouslyIgnoredProtocol")) {
+                self.assertSuperfluousIgnoreCommand(.varInstance("superfluousProperty"))
+            }
+        }
+    }
+
     // MARK: - Swift Testing
 
     #if canImport(Testing)

